sympy/diffgeom/diffgeom.py

```diff
@@ -130,110 +151,34 @@ class CoordSystem(Atom):
-
-    Calculate the jacobian of the polar to cartesian transformation:
-
-    >>> polar.jacobian(rect, [r, theta])
-    Matrix([
-    [cos(theta), -r*sin(theta)],
-    [sin(theta),  r*cos(theta)]])
-
-    Define a point using coordinates in one of the coordinate systems:
-
-    >>> p = polar.point([1, 3*pi/4])
-    >>> rect.point_to_coords(p)
-    Matrix([
-    [-sqrt(2)/2],
-    [ sqrt(2)/2]])
-
-    Define a basis scalar field (i.e. a coordinate function), that takes a
-    point and returns its coordinates. It is an instance of ``BaseScalarField``.
-
-    >>> rect.coord_function(0)(p)
-    -sqrt(2)/2
-    >>> rect.coord_function(1)(p)
-    sqrt(2)/2
-
-    Define a basis vector field (i.e. a unit vector field along the coordinate
-    line). Vectors are also differential operators on scalar fields. It is an
-    instance of ``BaseVectorField``.
-
-    >>> v_x = rect.base_vector(0)
-    >>> x = rect.coord_function(0)
-    >>> v_x(x)
-    1
-    >>> v_x(v_x(x))
-    0
-
-    Define a basis oneform field:
-
-    >>> dx = rect.base_oneform(0)
-    >>> dx(v_x)
-    1
-
-    If you provide a list of names the fields will print nicely:
-    - without provided names:
-
-    >>> x, v_x, dx
-    (rect_0, e_rect_0, drect_0)
-
-    - with provided names
+    >>> m = Manifold('M', 3)
+    >>> p = Patch('P', m)
 
-    >>> rect = CoordSystem('rect', patch, ['x', 'y'])
-    >>> rect.coord_function(0), rect.base_vector(0), rect.base_oneform(0)
-    (x, e_x, dx)
+    >>> rect = CoordSystem('rect', p)
+    >>> rect.name
+    'rect'
+    >>> rect.dim
+    3
 
     """
-    #  Contains a reference to the parent patch in order to be able to access
-    # other coordinate system charts.
     def __new__(cls, name, patch, names=None):
-        # names is not in args because it is related only to printing, not to
-        # identifying the CoordSystem instance.
         if not names:
             names = ['%s_%d' % (name, i) for i in range(patch.dim)]
         obj = super().__new__(cls)
         obj.name = name
         obj._names = tuple(str(i) for i in names)
         obj.patch = patch
-        obj.patch.coord_systems.append(obj)
-        obj.transforms = {}
-        # All the coordinate transformation logic is in this dictionary in the
-        # form of:
-        #  key = other coordinate system
-        #  value = tuple of  # TODO make these Lambda instances
-        #          - list of `Dummy` coordinates in this coordinate system
-        #          - list of expressions as a function of the Dummies giving
-        #          the coordinates in another coordinate system
+        obj.patch.coord_systems.append(obj) # deprecated
+        obj.transforms = _deprecated_dict(
+            "Using CoordSystem.transforms for storing transform relations",
+            "CoordSystemStructure",
+            19321,
+            "1.7",
+            {}
+        ) # deprecated
         obj._dummies = [Dummy(str(n)) for n in names]
         obj._dummy = Dummy()
```

Comment:
I was thinking of removing these dummies.
Have you tried anything successful?

Reply:
Yes. In fact, I have had a plan of removing these after this PR, as well. I'll make another PR and ping you there.

Reply:
`_dummy` is used in `BaseVectorField` in some strange way as an attribute of 'dummy function'. In my local repository, I have `DummyFunction` which is subclass of `Function`. I am planning to contribute that to SymPy, and after then we can deal with this issue.

Until then, removing these are out of this PR's scope so I'll mark this conversation as resolved.